private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/maven/MavenResolver.java

```diff
@@ -404,7 +404,7 @@ private List<Dependency> resolveArtifactsFromBoms(
       RepositorySystemSession session,
       List<RemoteRepository> repositories,
       List<Dependency> boms) {
-    Set<Dependency> managedDependencies = new HashSet<>();
```

Comment:
I think the only change you need for this is
```
Set<Dependency> managedDependencies = new LinkedHashSet<>();
```
So the Set keeps the order of the dependencies added to it.

Reply:
Yep, that was it, updated the PR.